teamcity: failed tests on master: acceptance/TestBuildInfo, acceptance/TestDockerC

The following tests appear to have failed:

[#551610](https://teamcity.cockroachdb.com/viewLog.html?buildId=551610):


```
--- FAIL: acceptance/TestBuildInfo: TestDockerC/Success/runMode=docker (0.000s)
Test ended in panic.

------- Stdout: -------
E180307 20:20:52.201899 1 cli/error.go:109  unable to connect or connection lost.

Please check the address and credentials such as certificates (if attempting to
communicate with a secure cluster).

initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
Error: unable to connect or connection lost.

Please check the address and credentials such as certificates (if attempting to
communicate with a secure cluster).

initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
Failed running "init"
non-zero exit code: 1
panic: test timed out after 30m0s

goroutine 137 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1240 +0xfc
created by time.goFunc
	/usr/local/go/src/time/sleep.go:172 +0x44

goroutine 1 [chan receive, 29 minutes]:
testing.(*T).Run(0xc420396a50, 0x1cdd452, 0xb, 0x1da2b40, 0x6cf901)
	/usr/local/go/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc420396690)
	/usr/local/go/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc420396690, 0xc420637da8)
	/usr/local/go/src/testing/testing.go:777 +0xd0
testing.runTests(0xc420564440, 0x2ab9c00, 0x1d, 0x1d, 0xbea03047b9d64e74)
	/usr/local/go/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc42003a100, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x171
github.com/cockroachdb/cockroach/pkg/acceptance.RunTests(0xc42003a100)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/test_acceptance.go:53 +0x48
github.com/cockroachdb/cockroach/pkg/acceptance.MainTest(0xc42003a100)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/test_acceptance.go:34 +0x2b
github.com/cockroachdb/cockroach/pkg/acceptance.TestMain(0xc42003a100)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/main_test.go:22 +0x2b
main.main()
	_testmain.go:96 +0x151

goroutine 5 [syscall, 30 minutes]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 37 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1095 +0xf1
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:582 +0xd9

goroutine 50 [chan receive, 30 minutes]:
github.com/cockroachdb/cockroach/pkg/acceptance.RunTests.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/test_acceptance.go:44 +0xa0
created by github.com/cockroachdb/cockroach/pkg/acceptance.RunTests
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/test_acceptance.go:40 +0x3a

goroutine 131 [chan receive, 29 minutes]:
testing.(*T).Run(0xc420396b40, 0x1cd6d74, 0x7, 0xc4202862a0, 0xc420396a50)
	/usr/local/go/src/testing/testing.go:825 +0x301
github.com/cockroachdb/cockroach/pkg/acceptance.TestDockerC(0xc420396a50)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/c_test.go:30 +0xea
testing.tRunner(0xc420396a50, 0x1da2b40)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 10 [select, 30 minutes, locked to thread]:
runtime.gopark(0x1da6580, 0x0, 0x1cd604e, 0x6, 0x18, 0x1)
	/usr/local/go/src/runtime/proc.go:291 +0x11a
runtime.selectgo(0xc4204b1750, 0xc420046240)
	/usr/local/go/src/runtime/select.go:392 +0xe50
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:549 +0x1f4
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1

goroutine 132 [chan receive, 29 minutes]:
testing.(*T).Run(0xc420396c30, 0x1ce1f98, 0xe, 0xc420858100, 0xc420858100)
	/usr/local/go/src/testing/testing.go:825 +0x301
github.com/cockroachdb/cockroach/pkg/acceptance.RunDocker(0xc420396b40, 0xc420858100)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_cluster.go:47 +0x4a
github.com/cockroachdb/cockroach/pkg/acceptance.testDocker(0x1f3a240, 0xc4200420c0, 0xc420396b40, 0x1, 0x1cd1471, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_docker.go:72 +0x114
github.com/cockroachdb/cockroach/pkg/acceptance.testDockerSingleNode(0x1f3a240, 0xc4200420c0, 0xc420396b40, 0x1cd1471, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_docker.go:107 +0xa0
github.com/cockroachdb/cockroach/pkg/acceptance.testDockerSuccess(0x1f3a240, 0xc4200420c0, 0xc420396b40, 0x1cd1471, 0x1, 0xc420768180, 0x3, 0x3)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_docker.go:57 +0xea
github.com/cockroachdb/cockroach/pkg/acceptance.TestDockerC.func1(0xc420396b40)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/c_test.go:31 +0x14b
testing.tRunner(0xc420396b40, 0xc4202862a0)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 157 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42045e320, 0x1f3a200, 0xc4208589c0)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 133 [IO wait, 29 minutes]:
internal/poll.runtime_pollWait(0x7fe2854adc90, 0x72, 0xc4209c2508)
	/usr/local/go/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc4204aee18, 0x72, 0xffffffffffffff00, 0x1f21420, 0x2acba18)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0x9b
internal/poll.(*pollDesc).waitRead(0xc4204aee18, 0xc4202e3000, 0x1, 0x200)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc4204aee00, 0xc4202e30a0, 0x1, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:157 +0x17d
net.(*netFD).Read(0xc4204aee00, 0xc4202e30a0, 0x1, 0x200, 0x8, 0x1ff, 0x8)
	/usr/local/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc42000e180, 0xc4202e30a0, 0x1, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:176 +0x6a
io.ReadAtLeast(0x7fe285389020, 0xc42000e180, 0xc4202e30a0, 0x1, 0x200, 0x1, 0x1c854e0, 0x0, 0x7fe285389020)
	/usr/local/go/src/io/io.go:309 +0x86
io.ReadFull(0x7fe285389020, 0xc42000e180, 0xc4202e30a0, 0x1, 0x200, 0x652de8, 0x3, 0x1b2cd60)
	/usr/local/go/src/io/io.go:327 +0x58
github.com/cockroachdb/cockroach/vendor/github.com/lib/pq.(*conn).ssl(0xc4202e3080, 0xc420769560)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/lib/pq/conn.go:1036 +0x184
github.com/cockroachdb/cockroach/vendor/github.com/lib/pq.DialOpen(0x1f27e00, 0x2d39880, 0xc42020d200, 0x1f2, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/lib/pq/conn.go:342 +0x5b1
github.com/cockroachdb/cockroach/vendor/github.com/lib/pq.Open(0xc42020d200, 0x1f2, 0x10000c420740280, 0x0, 0x1f2, 0xc42020d200)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/lib/pq/conn.go:251 +0x4d
github.com/cockroachdb/cockroach/vendor/github.com/lib/pq.(*Driver).Open(0x2d39880, 0xc42020d200, 0x1f2, 0x20, 0x252, 0x252, 0xc420740280)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/lib/pq/conn.go:45 +0x35
database/sql.dsnConnector.Connect(0xc42020d200, 0x1f2, 0x1f1edc0, 0x2d39880, 0x1f3a240, 0xc4200420c0, 0x0, 0x0, 0x100, 0x0)
	/usr/local/go/src/database/sql/sql.go:600 +0x45
database/sql.(*DB).conn(0xc42045e320, 0x1f3a240, 0xc4200420c0, 0xc4209c2c01, 0x792c3e, 0x18, 0x1da27e8)
	/usr/local/go/src/database/sql/sql.go:1103 +0x131
database/sql.(*DB).query(0xc42045e320, 0x1f3a240, 0xc4200420c0, 0x1d39e8e, 0x42, 0x0, 0x0, 0x0, 0x1, 0x100000000000000, ...)
	/usr/local/go/src/database/sql/sql.go:1437 +0x66
database/sql.(*DB).QueryContext(0xc42045e320, 0x1f3a240, 0xc4200420c0, 0x1d39e8e, 0x42, 0x0, 0x0, 0x0, 0xc42045e320, 0x0, ...)
	/usr/local/go/src/database/sql/sql.go:1419 +0xd2
database/sql.(*DB).Query(0xc42045e320, 0x1d39e8e, 0x42, 0x0, 0x0, 0x0, 0x0, 0x688c47, 0xc4202414e8)
	/usr/local/go/src/database/sql/sql.go:1433 +0x82
github.com/cockroachdb/cockroach/pkg/acceptance.StartCluster.func2(0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_cluster.go:182 +0x237
github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0xa7a358200, 0xc42074d2e0, 0xbe274e, 0x1f3a240)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:173 +0xcc
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x1f56240, 0xc420396c30, 0xc4209c32e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x55
github.com/cockroachdb/cockroach/pkg/acceptance.StartCluster(0x1f3a240, 0xc4200420c0, 0xc420396c30, 0x1cd1471, 0x1, 0xc4207681e0, 0x1, 0x1, 0x12a05f200, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_cluster.go:170 +0x726
github.com/cockroachdb/cockroach/pkg/acceptance.testDocker.func1(0xc420396c30)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_docker.go:80 +0x2a3
testing.tRunner(0xc420396c30, 0xc420858100)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 21 [select, 30 minutes]:
database/sql.(*DB).connectionOpener(0xc42020e3c0, 0x1f3a200, 0xc42050e140)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 22 [select, 30 minutes]:
database/sql.(*DB).connectionResetter(0xc42020e3c0, 0x1f3a200, 0xc42050e140)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 40 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42012c640, 0x1f3a200, 0xc420207c00)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 103 [select, 30 minutes]:
database/sql.(*DB).connectionResetter(0xc4201d66e0, 0x1f3a200, 0xc4207f61c0)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 149 [IO wait, 29 minutes]:
internal/poll.runtime_pollWait(0x7fe2854adbc0, 0x72, 0xc420799798)
	/usr/local/go/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc4204aec18, 0x72, 0xffffffffffffff00, 0x1f21420, 0x2acba18)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0x9b
internal/poll.(*pollDesc).waitRead(0xc4204aec18, 0xc4205be000, 0x1000, 0x1000)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc4204aec00, 0xc4205be000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:157 +0x17d
net.(*netFD).Read(0xc4204aec00, 0xc4205be000, 0x1000, 0x1000, 0xc420322408, 0xc4203d837d, 0x5)
	/usr/local/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc42000e440, 0xc4205be000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:176 +0x6a
net/http.(*persistConn).Read(0xc4208117a0, 0xc4205be000, 0x1000, 0x1000, 0xc420322408, 0x198c27d, 0x5)
	/usr/local/go/src/net/http/transport.go:1453 +0x136
bufio.(*Reader).fill(0xc4200a3d40)
	/usr/local/go/src/bufio/bufio.go:100 +0x11e
bufio.(*Reader).ReadSlice(0xc4200a3d40, 0x1c6490a, 0xc4203d82d0, 0x199, 0x0, 0x0, 0x186)
	/usr/local/go/src/bufio/bufio.go:341 +0x2c
net/http/internal.readChunkLine(0xc4200a3d40, 0x8, 0x432, 0xa, 0x433, 0x8)
	/usr/local/go/src/net/http/internal/chunked.go:122 +0x34
net/http/internal.(*chunkedReader).beginChunk(0xc42051bef0)
	/usr/local/go/src/net/http/internal/chunked.go:48 +0x32
net/http/internal.(*chunkedReader).Read(0xc42051bef0, 0xc420536601, 0x5ff, 0x5ff, 0xc420799b28, 0x66bc89, 0xc400000008)
	/usr/local/go/src/net/http/internal/chunked.go:93 +0x113
net/http.(*body).readLocked(0xc420858000, 0xc420536601, 0x5ff, 0x5ff, 0xc420530000, 0xc420799ef0, 0xc420799ca8)
	/usr/local/go/src/net/http/transfer.go:778 +0x61
net/http.(*body).Read(0xc420858000, 0xc420536601, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transfer.go:770 +0xdd
net/http.(*bodyEOFSignal).Read(0xc420858040, 0xc420536601, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:2187 +0xdc
encoding/json.(*Decoder).refill(0xc4203223c0, 0xc420799e0a, 0x9)
	/usr/local/go/src/encoding/json/stream.go:159 +0x132
encoding/json.(*Decoder).readValue(0xc4203223c0, 0x0, 0x0, 0x1c649a0)
	/usr/local/go/src/encoding/json/stream.go:134 +0x23d
encoding/json.(*Decoder).Decode(0xc4203223c0, 0x1a04c60, 0xc420241050, 0x0, 0x0)
	/usr/local/go/src/encoding/json/stream.go:63 +0x78
github.com/cockroachdb/cockroach/vendor/github.com/docker/docker/client.(*Client).Events.func1(0xc420438060, 0xc4204aea00, 0x0, 0x0, 0x0, 0x0, 0xc4201f0270, 0xc4207a0060, 0x1f3a200, 0xc4207f6240, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/docker/docker/client/events.go:54 +0x2da
created by github.com/cockroachdb/cockroach/vendor/github.com/docker/docker/client.(*Client).Events
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/docker/docker/client/events.go:26 +0x11d

goroutine 135 [select, 29 minutes]:
net/http.(*persistConn).readLoop(0xc4208117a0)
	/usr/local/go/src/net/http/transport.go:1717 +0x743
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1237 +0x95a

goroutine 117 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc420525cc0, 0x1f3a200, 0xc4207e6380)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 156 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42045e320, 0x1f3a200, 0xc4208589c0)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 99 [select, 30 minutes]:
database/sql.(*DB).connectionOpener(0xc4201d6500, 0x1f3a200, 0xc4207f6080)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 100 [select, 30 minutes]:
database/sql.(*DB).connectionResetter(0xc4201d6500, 0x1f3a200, 0xc4207f6080)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 102 [select, 30 minutes]:
database/sql.(*DB).connectionOpener(0xc4201d66e0, 0x1f3a200, 0xc4207f61c0)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 11 [select, 30 minutes]:
database/sql.(*DB).connectionOpener(0xc42012c3c0, 0x1f3a200, 0xc42080c080)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 12 [select, 30 minutes]:
database/sql.(*DB).connectionResetter(0xc42012c3c0, 0x1f3a200, 0xc42080c080)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 116 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc420525cc0, 0x1f3a200, 0xc4207e6380)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 105 [select, 29 minutes]:
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*DockerCluster).monitor.func1(0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/dockercluster.go:648 +0x2bd
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*DockerCluster).monitor(0xc4200fae00, 0x1f3a240, 0xc4200420c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/dockercluster.go:668 +0x7b
created by github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*DockerCluster).Start
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/dockercluster.go:689 +0x37d

goroutine 39 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42012c640, 0x1f3a200, 0xc420207c00)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 119 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42020e780, 0x1f3a200, 0xc4207e6440)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 120 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42020e780, 0x1f3a200, 0xc4207e6440)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 121 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42020eb40, 0x1f3a200, 0xc4207e64c0)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 122 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42020eb40, 0x1f3a200, 0xc4207e64c0)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 41 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42012d9a0, 0x1f3a200, 0xc420207c80)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 42 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42012d9a0, 0x1f3a200, 0xc420207c80)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 136 [select, 29 minutes]:
net/http.(*persistConn).writeLoop(0xc4208117a0)
	/usr/local/go/src/net/http/transport.go:1822 +0x14b
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1238 +0x97f

goroutine 108 [IO wait, 29 minutes]:
internal/poll.runtime_pollWait(0x7fe2854ad200, 0x72, 0xc4200819a8)
	/usr/local/go/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc42003a498, 0x72, 0xffffffffffffff00, 0x1f21420, 0x2acba18)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0x9b
internal/poll.(*pollDesc).waitRead(0xc42003a498, 0xc42017e000, 0x1000, 0x1000)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc42003a480, 0xc42017e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:157 +0x17d
net.(*netFD).Read(0xc42003a480, 0xc42017e000, 0x1000, 0x1000, 0x698d90, 0xc4204a1500, 0x4)
	/usr/local/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc42075e0f0, 0xc42017e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:176 +0x6a
net/http.(*persistConn).Read(0xc420842360, 0xc42017e000, 0x1000, 0x1000, 0xc420081b98, 0x645685, 0xc420796960)
	/usr/local/go/src/net/http/transport.go:1453 +0x136
bufio.(*Reader).fill(0xc4200a3ce0)
	/usr/local/go/src/bufio/bufio.go:100 +0x11e
bufio.(*Reader).Peek(0xc4200a3ce0, 0x1, 0x0, 0x0, 0x0, 0xc420797380, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0x3a
net/http.(*persistConn).readLoop(0xc420842360)
	/usr/local/go/src/net/http/transport.go:1601 +0x185
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1237 +0x95a

goroutine 109 [select, 29 minutes]:
net/http.(*persistConn).writeLoop(0xc420842360)
	/usr/local/go/src/net/http/transport.go:1822 +0x14b
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1238 +0x97f
--- FAIL: acceptance/TestDockerC (0.000s)
Test ended in panic.

--- FAIL: acceptance/TestDockerC/Success (0.000s)
Test ended in panic.

--- FAIL: acceptance/TestBuildInfo: TestDockerC/Success/runMode=docker (0.000s)
Test ended in panic.

------- Stdout: -------
E180307 20:20:52.201899 1 cli/error.go:109  unable to connect or connection lost.

Please check the address and credentials such as certificates (if attempting to
communicate with a secure cluster).

initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
Error: unable to connect or connection lost.

Please check the address and credentials such as certificates (if attempting to
communicate with a secure cluster).

initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
Failed running "init"
non-zero exit code: 1
panic: test timed out after 30m0s

goroutine 137 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1240 +0xfc
created by time.goFunc
	/usr/local/go/src/time/sleep.go:172 +0x44

goroutine 1 [chan receive, 29 minutes]:
testing.(*T).Run(0xc420396a50, 0x1cdd452, 0xb, 0x1da2b40, 0x6cf901)
	/usr/local/go/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc420396690)
	/usr/local/go/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc420396690, 0xc420637da8)
	/usr/local/go/src/testing/testing.go:777 +0xd0
testing.runTests(0xc420564440, 0x2ab9c00, 0x1d, 0x1d, 0xbea03047b9d64e74)
	/usr/local/go/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc42003a100, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x171
github.com/cockroachdb/cockroach/pkg/acceptance.RunTests(0xc42003a100)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/test_acceptance.go:53 +0x48
github.com/cockroachdb/cockroach/pkg/acceptance.MainTest(0xc42003a100)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/test_acceptance.go:34 +0x2b
github.com/cockroachdb/cockroach/pkg/acceptance.TestMain(0xc42003a100)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/main_test.go:22 +0x2b
main.main()
	_testmain.go:96 +0x151

goroutine 5 [syscall, 30 minutes]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 37 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1095 +0xf1
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:582 +0xd9

goroutine 50 [chan receive, 30 minutes]:
github.com/cockroachdb/cockroach/pkg/acceptance.RunTests.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/test_acceptance.go:44 +0xa0
created by github.com/cockroachdb/cockroach/pkg/acceptance.RunTests
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/test_acceptance.go:40 +0x3a

goroutine 131 [chan receive, 29 minutes]:
testing.(*T).Run(0xc420396b40, 0x1cd6d74, 0x7, 0xc4202862a0, 0xc420396a50)
	/usr/local/go/src/testing/testing.go:825 +0x301
github.com/cockroachdb/cockroach/pkg/acceptance.TestDockerC(0xc420396a50)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/c_test.go:30 +0xea
testing.tRunner(0xc420396a50, 0x1da2b40)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 10 [select, 30 minutes, locked to thread]:
runtime.gopark(0x1da6580, 0x0, 0x1cd604e, 0x6, 0x18, 0x1)
	/usr/local/go/src/runtime/proc.go:291 +0x11a
runtime.selectgo(0xc4204b1750, 0xc420046240)
	/usr/local/go/src/runtime/select.go:392 +0xe50
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:549 +0x1f4
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1

goroutine 132 [chan receive, 29 minutes]:
testing.(*T).Run(0xc420396c30, 0x1ce1f98, 0xe, 0xc420858100, 0xc420858100)
	/usr/local/go/src/testing/testing.go:825 +0x301
github.com/cockroachdb/cockroach/pkg/acceptance.RunDocker(0xc420396b40, 0xc420858100)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_cluster.go:47 +0x4a
github.com/cockroachdb/cockroach/pkg/acceptance.testDocker(0x1f3a240, 0xc4200420c0, 0xc420396b40, 0x1, 0x1cd1471, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_docker.go:72 +0x114
github.com/cockroachdb/cockroach/pkg/acceptance.testDockerSingleNode(0x1f3a240, 0xc4200420c0, 0xc420396b40, 0x1cd1471, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_docker.go:107 +0xa0
github.com/cockroachdb/cockroach/pkg/acceptance.testDockerSuccess(0x1f3a240, 0xc4200420c0, 0xc420396b40, 0x1cd1471, 0x1, 0xc420768180, 0x3, 0x3)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_docker.go:57 +0xea
github.com/cockroachdb/cockroach/pkg/acceptance.TestDockerC.func1(0xc420396b40)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/c_test.go:31 +0x14b
testing.tRunner(0xc420396b40, 0xc4202862a0)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 157 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42045e320, 0x1f3a200, 0xc4208589c0)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 133 [IO wait, 29 minutes]:
internal/poll.runtime_pollWait(0x7fe2854adc90, 0x72, 0xc4209c2508)
	/usr/local/go/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc4204aee18, 0x72, 0xffffffffffffff00, 0x1f21420, 0x2acba18)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0x9b
internal/poll.(*pollDesc).waitRead(0xc4204aee18, 0xc4202e3000, 0x1, 0x200)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc4204aee00, 0xc4202e30a0, 0x1, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:157 +0x17d
net.(*netFD).Read(0xc4204aee00, 0xc4202e30a0, 0x1, 0x200, 0x8, 0x1ff, 0x8)
	/usr/local/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc42000e180, 0xc4202e30a0, 0x1, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:176 +0x6a
io.ReadAtLeast(0x7fe285389020, 0xc42000e180, 0xc4202e30a0, 0x1, 0x200, 0x1, 0x1c854e0, 0x0, 0x7fe285389020)
	/usr/local/go/src/io/io.go:309 +0x86
io.ReadFull(0x7fe285389020, 0xc42000e180, 0xc4202e30a0, 0x1, 0x200, 0x652de8, 0x3, 0x1b2cd60)
	/usr/local/go/src/io/io.go:327 +0x58
github.com/cockroachdb/cockroach/vendor/github.com/lib/pq.(*conn).ssl(0xc4202e3080, 0xc420769560)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/lib/pq/conn.go:1036 +0x184
github.com/cockroachdb/cockroach/vendor/github.com/lib/pq.DialOpen(0x1f27e00, 0x2d39880, 0xc42020d200, 0x1f2, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/lib/pq/conn.go:342 +0x5b1
github.com/cockroachdb/cockroach/vendor/github.com/lib/pq.Open(0xc42020d200, 0x1f2, 0x10000c420740280, 0x0, 0x1f2, 0xc42020d200)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/lib/pq/conn.go:251 +0x4d
github.com/cockroachdb/cockroach/vendor/github.com/lib/pq.(*Driver).Open(0x2d39880, 0xc42020d200, 0x1f2, 0x20, 0x252, 0x252, 0xc420740280)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/lib/pq/conn.go:45 +0x35
database/sql.dsnConnector.Connect(0xc42020d200, 0x1f2, 0x1f1edc0, 0x2d39880, 0x1f3a240, 0xc4200420c0, 0x0, 0x0, 0x100, 0x0)
	/usr/local/go/src/database/sql/sql.go:600 +0x45
database/sql.(*DB).conn(0xc42045e320, 0x1f3a240, 0xc4200420c0, 0xc4209c2c01, 0x792c3e, 0x18, 0x1da27e8)
	/usr/local/go/src/database/sql/sql.go:1103 +0x131
database/sql.(*DB).query(0xc42045e320, 0x1f3a240, 0xc4200420c0, 0x1d39e8e, 0x42, 0x0, 0x0, 0x0, 0x1, 0x100000000000000, ...)
	/usr/local/go/src/database/sql/sql.go:1437 +0x66
database/sql.(*DB).QueryContext(0xc42045e320, 0x1f3a240, 0xc4200420c0, 0x1d39e8e, 0x42, 0x0, 0x0, 0x0, 0xc42045e320, 0x0, ...)
	/usr/local/go/src/database/sql/sql.go:1419 +0xd2
database/sql.(*DB).Query(0xc42045e320, 0x1d39e8e, 0x42, 0x0, 0x0, 0x0, 0x0, 0x688c47, 0xc4202414e8)
	/usr/local/go/src/database/sql/sql.go:1433 +0x82
github.com/cockroachdb/cockroach/pkg/acceptance.StartCluster.func2(0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_cluster.go:182 +0x237
github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0xa7a358200, 0xc42074d2e0, 0xbe274e, 0x1f3a240)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:173 +0xcc
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x1f56240, 0xc420396c30, 0xc4209c32e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x55
github.com/cockroachdb/cockroach/pkg/acceptance.StartCluster(0x1f3a240, 0xc4200420c0, 0xc420396c30, 0x1cd1471, 0x1, 0xc4207681e0, 0x1, 0x1, 0x12a05f200, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_cluster.go:170 +0x726
github.com/cockroachdb/cockroach/pkg/acceptance.testDocker.func1(0xc420396c30)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/util_docker.go:80 +0x2a3
testing.tRunner(0xc420396c30, 0xc420858100)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 21 [select, 30 minutes]:
database/sql.(*DB).connectionOpener(0xc42020e3c0, 0x1f3a200, 0xc42050e140)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 22 [select, 30 minutes]:
database/sql.(*DB).connectionResetter(0xc42020e3c0, 0x1f3a200, 0xc42050e140)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 40 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42012c640, 0x1f3a200, 0xc420207c00)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 103 [select, 30 minutes]:
database/sql.(*DB).connectionResetter(0xc4201d66e0, 0x1f3a200, 0xc4207f61c0)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 149 [IO wait, 29 minutes]:
internal/poll.runtime_pollWait(0x7fe2854adbc0, 0x72, 0xc420799798)
	/usr/local/go/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc4204aec18, 0x72, 0xffffffffffffff00, 0x1f21420, 0x2acba18)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0x9b
internal/poll.(*pollDesc).waitRead(0xc4204aec18, 0xc4205be000, 0x1000, 0x1000)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc4204aec00, 0xc4205be000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:157 +0x17d
net.(*netFD).Read(0xc4204aec00, 0xc4205be000, 0x1000, 0x1000, 0xc420322408, 0xc4203d837d, 0x5)
	/usr/local/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc42000e440, 0xc4205be000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:176 +0x6a
net/http.(*persistConn).Read(0xc4208117a0, 0xc4205be000, 0x1000, 0x1000, 0xc420322408, 0x198c27d, 0x5)
	/usr/local/go/src/net/http/transport.go:1453 +0x136
bufio.(*Reader).fill(0xc4200a3d40)
	/usr/local/go/src/bufio/bufio.go:100 +0x11e
bufio.(*Reader).ReadSlice(0xc4200a3d40, 0x1c6490a, 0xc4203d82d0, 0x199, 0x0, 0x0, 0x186)
	/usr/local/go/src/bufio/bufio.go:341 +0x2c
net/http/internal.readChunkLine(0xc4200a3d40, 0x8, 0x432, 0xa, 0x433, 0x8)
	/usr/local/go/src/net/http/internal/chunked.go:122 +0x34
net/http/internal.(*chunkedReader).beginChunk(0xc42051bef0)
	/usr/local/go/src/net/http/internal/chunked.go:48 +0x32
net/http/internal.(*chunkedReader).Read(0xc42051bef0, 0xc420536601, 0x5ff, 0x5ff, 0xc420799b28, 0x66bc89, 0xc400000008)
	/usr/local/go/src/net/http/internal/chunked.go:93 +0x113
net/http.(*body).readLocked(0xc420858000, 0xc420536601, 0x5ff, 0x5ff, 0xc420530000, 0xc420799ef0, 0xc420799ca8)
	/usr/local/go/src/net/http/transfer.go:778 +0x61
net/http.(*body).Read(0xc420858000, 0xc420536601, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transfer.go:770 +0xdd
net/http.(*bodyEOFSignal).Read(0xc420858040, 0xc420536601, 0x5ff, 0x5ff, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:2187 +0xdc
encoding/json.(*Decoder).refill(0xc4203223c0, 0xc420799e0a, 0x9)
	/usr/local/go/src/encoding/json/stream.go:159 +0x132
encoding/json.(*Decoder).readValue(0xc4203223c0, 0x0, 0x0, 0x1c649a0)
	/usr/local/go/src/encoding/json/stream.go:134 +0x23d
encoding/json.(*Decoder).Decode(0xc4203223c0, 0x1a04c60, 0xc420241050, 0x0, 0x0)
	/usr/local/go/src/encoding/json/stream.go:63 +0x78
github.com/cockroachdb/cockroach/vendor/github.com/docker/docker/client.(*Client).Events.func1(0xc420438060, 0xc4204aea00, 0x0, 0x0, 0x0, 0x0, 0xc4201f0270, 0xc4207a0060, 0x1f3a200, 0xc4207f6240, ...)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/docker/docker/client/events.go:54 +0x2da
created by github.com/cockroachdb/cockroach/vendor/github.com/docker/docker/client.(*Client).Events
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/docker/docker/client/events.go:26 +0x11d

goroutine 135 [select, 29 minutes]:
net/http.(*persistConn).readLoop(0xc4208117a0)
	/usr/local/go/src/net/http/transport.go:1717 +0x743
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1237 +0x95a

goroutine 117 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc420525cc0, 0x1f3a200, 0xc4207e6380)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 156 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42045e320, 0x1f3a200, 0xc4208589c0)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 99 [select, 30 minutes]:
database/sql.(*DB).connectionOpener(0xc4201d6500, 0x1f3a200, 0xc4207f6080)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 100 [select, 30 minutes]:
database/sql.(*DB).connectionResetter(0xc4201d6500, 0x1f3a200, 0xc4207f6080)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 102 [select, 30 minutes]:
database/sql.(*DB).connectionOpener(0xc4201d66e0, 0x1f3a200, 0xc4207f61c0)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 11 [select, 30 minutes]:
database/sql.(*DB).connectionOpener(0xc42012c3c0, 0x1f3a200, 0xc42080c080)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 12 [select, 30 minutes]:
database/sql.(*DB).connectionResetter(0xc42012c3c0, 0x1f3a200, 0xc42080c080)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 116 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc420525cc0, 0x1f3a200, 0xc4207e6380)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 105 [select, 29 minutes]:
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*DockerCluster).monitor.func1(0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/dockercluster.go:648 +0x2bd
github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*DockerCluster).monitor(0xc4200fae00, 0x1f3a240, 0xc4200420c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/dockercluster.go:668 +0x7b
created by github.com/cockroachdb/cockroach/pkg/acceptance/cluster.(*DockerCluster).Start
	/go/src/github.com/cockroachdb/cockroach/pkg/acceptance/cluster/dockercluster.go:689 +0x37d

goroutine 39 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42012c640, 0x1f3a200, 0xc420207c00)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 119 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42020e780, 0x1f3a200, 0xc4207e6440)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 120 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42020e780, 0x1f3a200, 0xc4207e6440)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 121 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42020eb40, 0x1f3a200, 0xc4207e64c0)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 122 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42020eb40, 0x1f3a200, 0xc4207e64c0)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 41 [select, 29 minutes]:
database/sql.(*DB).connectionOpener(0xc42012d9a0, 0x1f3a200, 0xc420207c80)
	/usr/local/go/src/database/sql/sql.go:935 +0x119
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:634 +0x178

goroutine 42 [select, 29 minutes]:
database/sql.(*DB).connectionResetter(0xc42012d9a0, 0x1f3a200, 0xc420207c80)
	/usr/local/go/src/database/sql/sql.go:948 +0x12a
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:635 +0x1ae

goroutine 136 [select, 29 minutes]:
net/http.(*persistConn).writeLoop(0xc4208117a0)
	/usr/local/go/src/net/http/transport.go:1822 +0x14b
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1238 +0x97f

goroutine 108 [IO wait, 29 minutes]:
internal/poll.runtime_pollWait(0x7fe2854ad200, 0x72, 0xc4200819a8)
	/usr/local/go/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc42003a498, 0x72, 0xffffffffffffff00, 0x1f21420, 0x2acba18)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:85 +0x9b
internal/poll.(*pollDesc).waitRead(0xc42003a498, 0xc42017e000, 0x1000, 0x1000)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc42003a480, 0xc42017e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/internal/poll/fd_unix.go:157 +0x17d
net.(*netFD).Read(0xc42003a480, 0xc42017e000, 0x1000, 0x1000, 0x698d90, 0xc4204a1500, 0x4)
	/usr/local/go/src/net/fd_unix.go:202 +0x4f
net.(*conn).Read(0xc42075e0f0, 0xc42017e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:176 +0x6a
net/http.(*persistConn).Read(0xc420842360, 0xc42017e000, 0x1000, 0x1000, 0xc420081b98, 0x645685, 0xc420796960)
	/usr/local/go/src/net/http/transport.go:1453 +0x136
bufio.(*Reader).fill(0xc4200a3ce0)
	/usr/local/go/src/bufio/bufio.go:100 +0x11e
bufio.(*Reader).Peek(0xc4200a3ce0, 0x1, 0x0, 0x0, 0x0, 0xc420797380, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0x3a
net/http.(*persistConn).readLoop(0xc420842360)
	/usr/local/go/src/net/http/transport.go:1601 +0x185
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1237 +0x95a

goroutine 109 [select, 29 minutes]:
net/http.(*persistConn).writeLoop(0xc420842360)
	/usr/local/go/src/net/http/transport.go:1822 +0x14b
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:1238 +0x97f
--- FAIL: acceptance/TestDockerC (0.000s)
Test ended in panic.

--- FAIL: acceptance/TestDockerC/Success (0.000s)
Test ended in panic.


```

Please assign, take a look and update the issue accordingly.

--- a/pkg/acceptance/cluster/dockercluster.go
+++ b/pkg/acceptance/cluster/dockercluster.go
@@ -546,36 +546,25 @@ func (l *DockerCluster) RunInitCommand(ctx context.Context, nodeIdx int) {
 
 	log.Infof(ctx, "trying to initialize via %v", containerConfig.Cmd)
 	// This is called early in the bootstrap sequence, and the node may not have
-	// opened its ports yet. Retry appropriately.
-	//
-	// TODO(tschottdorf): production code wouldn't do it like this. Instead,
-	// you'd wait until the healthz endpoint indicates that the init command
-	// should be tried. Perhaps these subtleties should simply be folded into
-	// `./cockroach init` as users are likely to get this wrong.
-	//
-	// See #19791.
+	// opened its ports yet. Wait for the health endpoint to become available,
+	// because we only get one shot at running the init command. (Retrying the
+	// init command is dangerous [0].)
+	// [0]: https://github.com/cockroachdb/cockroach/pull/19753#issuecomment-341561452
 	maybePanic(retry.ForDuration(time.Minute, func() error {
-		err := l.OneShot(
-			ctx, defaultImage, types.ImagePullOptions{}, containerConfig, container.HostConfig{}, "init-command",
-		)
-
-		if err != nil {
-			// Run a health check in the hope that Init failed because a
-			// previously issued Init actually went through just fine.
-			url := l.URL(ctx, nodeIdx) + "/health"
-			resp, httpErr := HTTPClient.Get(url)
-			if httpErr != nil {
-				return errors.Wrap(err, httpErr.Error())
-			}
-			defer resp.Body.Close()
-
-			if resp.StatusCode != http.StatusOK {
-				log.Warning(ctx, "init failed and server not healthy; retrying")
-				return err
-			}
+		url := l.URL(ctx, nodeIdx) + "/health"
+		resp, httpErr := HTTPClient.Get(url)
+		if httpErr != nil {
+			return httpErr
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			log.Warning(ctx, "server not healthy; retrying")
+			return errors.New("server not healthy")
 		}
 		return nil
 	}))
+	maybePanic(l.OneShot(ctx, defaultImage, types.ImagePullOptions{},
+		containerConfig, container.HostConfig{}, "init-command"))
 	log.Info(ctx, "cluster successfully initialized")
 }
 

